### ISSUE TITLE
Single is transform into double click since Macos catalina

### ISSUE THREAD

**Issue by amasson84** [kivy/kivy#6564]
Since I updated to macOS Catalina I have a bug with the trackpad of my mac

The single-click is replaced with a double click. I have to restart my script many times (some time more than 10 times) in order to have the single click correct.
I didn't notice this problem with an external mouse.

Let me know what I can provide as informations in order to correct this bug.
I use 
Config.set('input', 'mouse', 'mouse,multitouch_on_demand')
at the begining of my script

### Versions

* Python: 3.7.4
* OS: macos catalina
* Kivy: from master github
* Kivy installation method: pip

### Code and Logs
[INFO   ] [Logger      ] Record log in /Users/antoine/.kivy/logs/kivy_19-10-21_62.txt
[INFO   ] [Kivy        ] v2.0.0.dev0, git-Unknown, 20191019
[INFO   ] [Kivy        ] Installed at "/Users/antoine/Documents/Miro/MIRO-Lab-Software/envs/lib/python3.7/site-packages/kivy/__init__.py"
[INFO   ] [Python      ] v3.7.4 (default, Aug 13 2019, 15:17:50) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
[INFO   ] [Python      ] Interpreter at "/Users/antoine/Documents/Miro/MIRO-Lab-Software/envs/bin/python"
[INFO   ] [Logger      ] Purge log fired. Analysing...
[INFO   ] [Logger      ] Purge 21 log files
[INFO   ] [Logger      ] Purge finished!
[INFO   ] [Factory     ] 184 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_imageio, img_dds, img_sdl2, img_pil, img_gif (img_ffpyplayer ignored)
[INFO   ] [Text        ] Provider: sdl2(['text_pango'] ignored)
[INFO   ] [Window      ] Provider: sdl2
[INFO   ] [GL          ] Using the "OpenGL ES 2" graphics system
[INFO   ] [GL          ] Backend used <sdl2>
[INFO   ] [GL          ] OpenGL version <b'2.1 ATI-3.0.68'>
[INFO   ] [GL          ] OpenGL vendor <b'ATI Technologies Inc.'>
[INFO   ] [GL          ] OpenGL renderer <b'AMD Radeon Pro 560X OpenGL Engine'>
[INFO   ] [GL          ] OpenGL parsed version: 2, 1
[INFO   ] [GL          ] Shading version <b'1.20'>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <16>
[INFO   ] [Window      ] auto add sdl2 input provider
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
 2019-10-21 17:05:37,036 - gui_fct       DEBUG    class plot init
[INFO   ] [GL          ] NPOT texture support is available

**Comment by sbrother** [comment]
I'm running into this as well. It seems to happen:

- 100% of the time when using the trackpad
- 70% of the time when using an external non-Apple mouse but running from within PyCharm
- 20% of the time when using an external Apple magic mouse running from the terminal
- 0% of the time when using an external non-Apple mouse running from the terminal

Did you get anywhere debugging it?

**Comment by sbrother** [comment]
This actually seems to be completely fixed if you install kivy from source from master

**Comment by tshirtman** [comment]
if it persists, make sure to run with log_level=debug (in config.ini) to make sure there are no multiple touch providers for the same device (mouse).

**Comment by amasson84** [comment]
I still have this problem with the master branch, for me, I have the problem also with non-apple devices.
What I noticed is this problem appends if the mouse is moving when I start my python script (during the starting process)

**Comment by sbrother** [comment]
Yeah - this is a tough one since it's nondeterministic. I haven't seen it since I updated Kivy but I will report back if/when I do again. I wonder if it's some sort of race condition. I went down a rabbit hole looking for bug reports in SDL2 but couldn't find anything.

**Comment by sbrother** [comment]
And, it just manifested for me again, once in 50-100 runs.

**Comment by tshirtman** [comment]
that's a bit of a wild guess, but if it's a race condition, you never know:
- the logs are purged randomly with a 1 in 20 chance, at startup
- in the log that was shared up there, it was a case where the logs were purged.

I've no idea yet of the relationship between both, but if you can check that that was the case when you had the case for you as well, or if you force it to happen (by commenting out this block in a local copy of kivy https://github.com/kivy/kivy/blob/master/kivy/logger.py#L153-L156 and running it), if the correlation is confirmed, or that it at least increase the chances of it happening, that might help understanding.
